### PR TITLE
Single update task for resolver preview

### PR DESCRIPTION
**Pull request by robinschmid**
This improves the updating of the preview - but it also bugs out somehow. 
It runs on a different thread and should update the chart on jfx

### CODE REVIEW

**Review by SteffenHeu — CHANGES_REQUESTED** [review]
i think youre doing the chart changes outside of the java fx thread

**Comment by robinschmid** [comment]
So the idea is to update everything outside javafx and then fire the chart change event on javafx - thats done by the 
`chart.applyWithNotifyChanges`

**Comment by SteffenHeu** [comment]
But you're doing changes to an java fx object outside the java fx thread. that will only lead to problems.

**Comment by SteffenHeu** [comment]
why not calculate the datasets in one thread and do all the changes when youre finished?

**Comment by robinschmid** [comment]
That might work. I mean we are not changing the ChartView only the chart. but thats the same in the end I guess... 
So I double-checked chart.setNotify(false) actually cancels all updates. And plot updates are just passed up to the chart but not to the chartview.

**Comment by robinschmid** [comment]
Found it. All changes to the renderer need to use notify=false
Also need different renderers for each feature

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_chromatogramdeconvolution/FeatureResolverSetupDialog.java [hunk 1]
```diff
@@ -189,98 +201,150 @@ public ModularFeature fromString(String string) {
...
-                + rtFormat.format(
-                    series.getSpectra().get(series.getNumberOfValues() - 1).getRetentionTime())
-                + " min", new SimpleObjectProperty<>(palette.get(resolvedFeatureCounter++))));
-            MZmineCore.runLater(() -> chart.addDataset(ds, shapeRenderer));
+      @Override
+      public void run() {
+        setStatus(TaskStatus.PROCESSING);
+        chart.applyWithNotifyChanges(false, true, () -> {
+          logger.finest("Updating feature resolving preview");
+          chart.removeAllDatasets();
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/dataprocessing/featdet_chromatogramdeconvolution/FeatureResolverSetupDialog.java:234]: is actually fine. THe problem was in changing the renderer and reusing the renderer for all features

File: src/main/java/io/github/mzmine/gui/chartbasics/simplechart/SimpleXYChart.java [hunk 2]
```diff
@@ -201,17 +198,24 @@ private void initLabelListeners() {
...
+    setNotifyChange(false);
     // jfreechart renderers dont check if the value actually changed and notify either way
     if (renderer.getDefaultItemLabelsVisible() != isItemLabelsVisible()) {
-      renderer.setDefaultItemLabelsVisible(isItemLabelsVisible());
+      renderer.setDefaultItemLabelsVisible(isItemLabelsVisible(), false);
     }
     if (renderer.getDefaultSeriesVisibleInLegend() != isLegendItemsVisible()) {
-      renderer.setDefaultItemLabelsVisible(isLegendItemsVisible());
+      renderer.setDefaultSeriesVisibleInLegend(isLegendItemsVisible(), false);
     }
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/gui/chartbasics/simplechart/SimpleXYChart.java:209]: Reusing the same renderer for all was the problem and we need to set notify to false while changing them.

**Comment by SteffenHeu** [comment]
looks goodto me

**Pull request merged**